Added compilation support for sim annealing

AzAINN opened the pull request:
- added simulated annealing support on SimuQ for backend addition in QHDOPT

AzAINN committed:
Added compilation support for sim annealing

AzAINN commented:
@bernalde

bernalde commented on src/simuq/dwave/__init__.py:
```diff
@@ -1 +1,2 @@
 from simuq.dwave.dwave_provider import DWaveProvider
+from simuq.dwave.dwave_sim_provider import DWaveSimProvider
```
This class name is not clear enough to denote that it is simulated annealing; it gives the impression that it was a DWave quantum annealing simulator, which is not.

bernalde commented on src/simuq/dwave/dwave_sim_provider.py:
File name suggests it is not simulated annealing but a DWave simulator. Change please

AzAINN replied:
Removing the duplicate file in the PR

bernalde commented on src/simuq/dwave/dwave_sim_provider.py:
```diff
@@ -0,0 +1,132 @@
+    def compile(self,
+                qs,
+                anneal_schedule=[[0, 0], [20, 1]],
+                chain_strength=0.5,
```
chain_strength is not relevant for simulated annealing so you should remove it

bernalde commented on src/simuq/dwave/dwave_sim_provider.py:
```diff
@@ -0,0 +1,132 @@
+
+    def compile(self,
+                qs,
+                anneal_schedule=[[0, 0], [20, 1]],
```
The annealing schedule does not correspond to simulated annealing. There are other parameters such as the beta_range, beta_schedule_type, num_sweeps, etc. See if you can implement it maybe as kwargs. Details here https://github.com/dwavesystems/dwave-neal

AzAINN replied:
I have implemented them at compile time and also at run time, so if need be, they can be overridden per call. Do you think this is clean, or should I just have it at the compilation level

bernalde commented on src/simuq/dwave/dwave_sim_provider.py:
```diff
@@ -0,0 +1,132 @@
+        
+        # Use SimulatedAnnealingSampler instead of DWaveSampler
+        sampler = SimulatedAnnealingSampler()
+        h, J, anneal_schedule = self.prog
```
Here you would pass the arguments

bernalde commented on src/simuq/dwave/dwave_sim_provider.py:
```diff
@@ -0,0 +1,132 @@
+        
+        self.samples = list(response.samples())
+        
+        # Simulated annealing doesn't have QPU timing, so setting these to 0
```
It does not, but it may have other timing parameters. Better to pass those as they might be relevant

bernalde reviewed (changes requested):
There are some changes to be made regarding naming and using the right parameters for simulated annealing

AzAINN committed:
Addressed review feedback

AzAINN committed:
file name change

AzAINN commented:
@bernalde 
- changed file name
- changed class name
- added kwargs to handle additional SA specific parameters like beta_range, num_sweeps, beta_schedule, etc...
- kwargs at the compile() level and at the run() level
- Clean comments and removed unused lines that were copied from the dwave_provider.py file

AzAINN committed:
Delete src/simuq/dwave/dwave_sim_provider.py
